src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -151,16 +146,17 @@ export class InstructorCoursesPageComponent implements OnInit {
       this.statusMessageService.showErrorMessage(`Course ${courseId} is not found!`);
       return;
     }
-    const paramMap: { [key: string]: string } = {
-      courseid: courseId,
-      user: this.user,
-    };
-    this.httpRequestService.get('/course/stats', paramMap).subscribe((resp: CourseStats) => {
+    this.studentService.getStudentsFromCourse(courseId).subscribe((students: Students) => {
       this.courseStats[courseId] = {
-        sections: resp.sectionsTotal,
-        teams: resp.teamsTotal,
-        students: resp.studentsTotal,
-        unregistered: resp.unregisteredTotal,
+        sections: students.students.map((value: Student) => value.sectionName)
```

Comment:
That's a Interesting way to remove duplications 😃 

Try https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set ?